polarion/xml.py

```diff
@@ -0,0 +1,300 @@
+import xml.etree.ElementTree as ET
+from .polarion import Polarion
+from .record import Record
+from datetime import datetime
+import logging, json
+logger = logging.getLogger(__name__)
+
+class Config:
+    """
+    Config structure for xml importer.
+    """
+    XML_FILE='xml_file'                                 # Xml file to import
+    URL='url'                                           # Polarion url such as http://hostname/polarion
+    USERNAME='username'                                 # username to log to polarion
+    PASSWORD='password'                                 # password to log to polarion
+    TOKEN='token'                                       # token to log to polarion
+    PROJECT_ID='project_id'                             # id of the project
+    TESTRUN_ID='testrun_id'                             # id of the test run (if you want to use and existing one)
+    TESTRUN_ID_GENERATOR='testrun_id_generator'         # function that generate the testrun_id (config as parameter) if testrun_id is not provided. If not set: f'unit-{datetime.now}'
+    TESTRUN_TITLE='testrun_title'                       # title of the test run if testrun is created. If not set 'New unit test run'
+    TESTRUN_TYPE='testrun_type'                         # type of the test run if testrun is created. If not set 'xUnit Test Manual Upload'
+    TESTRUN_COMMENT='testrun_comment'                   # test run comment to add if set.
+    SKIP_MISSING_TESTCASE='skip_missing_testcase'       # if set to True, skip result on unknown test cases
+    ATTRIBUTES = [
+        XML_FILE, URL, USERNAME, PASSWORD, TOKEN, 
+        PROJECT_ID, TESTRUN_ID, TESTRUN_ID_GENERATOR,
+        TESTRUN_TITLE, TESTRUN_TYPE, TESTRUN_COMMENT]
+    MANDATORY = [XML_FILE, PROJECT_ID]  # and also either user/password or token
```

Comment:
SKIP_MISSING_TESTCASE is missing here. It would seem there need to be more MANDATORY items.

Reply:
Ok for SKIP_MISSING_TESTCASE missing as ATTRIBUTES.
I add URL as mandatory. Token or user/password are manually check as put in comment.
I do not see any other mandatory to check:
- TESTRUN_ID is optional. If not set it created a new one at each call
- TESTRUN_ID_GENERATOR has a default value
- TESTRUN_TITLE has a default value
- TESTRUN_TYPE has a default value
- TESTRUN_COMMENT is optional
- SKIP_MISSING_TESTCASE is optional